src/index.ts

```diff
@@ -86,59 +86,62 @@ const server = createServer(async (request, response) => {
         "<-- END OF LIST OF MODELS -->",
       ].join("\n"),
     },
-     ...compatibilityPayload.messages,
+    ...compatibilityPayload.messages,
   ];
 
   console.time("tool-call");
   const toolCaller = await capiClient.chat.completions.create({
     stream: false,
     model: "gpt-4o",
     messages: toolCallMessages,
-    tool_choice: "auto",
     tools: functions.map((f) => f.tool),
   });
   console.timeEnd("tool-call");
 
+  const [functionToCall] = getFunctionCalls(
+    // @ts-expect-error - type error due to Copilot/OpenAI SDKs interop, I'll look into it ~@gr2m
+    toolCaller.choices[0]
```

Comment:
this is not a straight forward fix. I will look into it, your call if you want to merge it or not